cpp/graph_algorithms_module/algorithms/algorithms.hpp

```diff
@@ -0,0 +1,119 @@
+
+#pragma once
+
+#include "data_structures/graph.hpp"
+
+namespace algorithms {
+
+/// Returns true if graph represented by edges is bipartite.
+/// Otherwise false.
+bool IsGraphBipartite(const std::vector<std::pair<uint32_t, uint32_t>> &edges);
+
+/// Returns true if graph is bipartite.
+/// Otherwise returns false.
+bool IsGraphBipartite(const graphdata::GraphView &G);
+
+/// Returns the size of the maximum matching in a given bipartite graph. The
+/// graph is defined by the sizes of two disjoint sets of nodes and by a set of
+/// edges between those two sets of nodes. The nodes in both sets should be
+/// indexed from 1 to `set_size`.
+///
+/// The algorithm runs in O(|V|*|E|) time where V represents a set of nodes and
+/// E represents a set of edges.
+size_t BipartiteMatching(const graphdata::GraphView &G);
+
+/// Returns the size of the maximum matching in a given bipartite graph. The
+/// graph is defined by the sizes of two disjoint sets of nodes and by a set of
+/// edges between those two sets of nodes. The nodes in both sets should be
+/// indexed from 1 to `set_size`.
+///
+/// The algorithm runs in O(|V|*|E|) time where V represents a set of nodes and
+/// E represents a set of edges.
+size_t BipartiteMatching(
+    const std::vector<std::pair<uint32_t, uint32_t>> &edges);
+
+/// Returns a list of edges of an undirected graph whose removal would
+/// add another connected component to G.
+///
+/// The algorithm runs in O(|E| + |V|) time where E represents a set
+/// of edges and V represents a set of vertices.
+std::vector<graphdata::Edge> GetBridges(const graphdata::GraphView &G);
+
+/// Returns a list of Biconnected Components (BCCs) of an undirected graph G
+/// Each BCC is represented by a list of edges that are inside that component.
+///
+/// The implemented algorithm is authored by Hopcroft and Tarjan and runs in
+/// O(|E| + |V|) time, where E represents a set of edges and V represents a
+/// set of vertices.
+std::vector<std::vector<graphdata::Edge>> GetBiconnectedComponents(
+    const graphdata::GraphView &G);
+
+/// Returns a list of all simple cycles of an undirected graph G. Each simple
+/// cycle is represented by a list of nodes in cycle order. For example, the
+/// cycle below would be represented by a list {1, 2, 3, 4}.
+///
+/// (1)--(2)
+///  |    |
+/// (4)--(3)
+///
+/// There is no other guarantee on the order of the nodes in a cycle.
+///
+/// The implemented algorithm (Gibb) is described in the 1982 MIT report called
+/// "Algorithmic approaches to circuit enumeration problems and applications"
+/// by Boon Chai Lee.
+///
+/// The problem itself is not solvable in polynomial time. The bulk of the
+/// algorithm's runtime is spent on finding all subsets of fundamental cycles
+/// which takes about O(2^(|E|-|V|+1)) time, where E represents a set of
+/// edges and V represents a set of vertices.
+///
+/// TODO(ipaljak): Optimize the algorithm by decomposition into biconnected
+///                components.
```

Comment:
Implement TODO or update who is owning this TODO (do the same for the rest of TODOs as well).